File: 2015/01_Not_Quite_Lisp/part_1/main.c
/**
 * Author: happ_2h https://github.com/happ-2h
 *
 * Solution for Advent of Code 2015 Day 1.1
 */

#include <stdio.h>

#define BUF_LEN 0x2000

int main(int argc, char const *argv[]) {
  FILE* ifile = fopen("../input.txt", "r");

  if (ifile) {
    int floor = 0; // Start at floor 0

    char ch = ' ';
    while((ch = fgetc(ifile)) != EOF)
      /*
        '(' = ascend floor
        ')' = descend floor
      */
      floor = ch == '(' ? floor + 1 : floor - 1;

    fprintf(stdout, "%d\n", floor);

    fclose(ifile);
    ifile = NULL;
  }
  else {
    perror("Failed to read input.txt");
    return 1;
  }

  return 0;
}

File: 2015/01_Not_Quite_Lisp/part_1/main.f90
!!!
! Author: happ_2h https://github.com/happ-2h
!
! Solution for Advent of Code Day 1.1
!!!

program notquitelisp
  implicit none

  integer :: floor, i, opstat
  character(len=7000) :: cc

  floor = 0
  i = 1

  open(1, file="../input.txt", status="old", action="read", iostat=opstat)
  if (opstat > 0) stop "Cannot open file input.txt"

    read(1, *) cc

    do while (i .ne. len(cc)+1)
      if (cc(i:i) .eq. "(") then ! Ascend
        floor = floor + 1
      else                       ! Descend
        floor = floor - 1
      end if
      i = i + 1                  ! Get next character
    end do

    write(*,'(i3)') floor

  close(1)

end program notquitelisp